mlflow/system_metrics/metrics/network_monitor.py

```diff
@@ -7,8 +7,29 @@
 
 
 class NetworkMonitor(BaseMetricsMonitor):
+    def __init__(self):
+        super().__init__()
+        self._set_initial_metrics()
+
+    def _set_initial_metrics(self):
+        # Set initial network usage metrics. `psutil.net_io_counters()` counts the stats since the
+        # system boot, so to set network usage metrics as 0 when we start logging, we need to keep
+        # the initial network usage metrics.
+        network_usage = psutil.net_io_counters()
+        self._initial_receive_megabytes = network_usage.bytes_recv / 1e6
+        self._initial_transmit_megabytes = network_usage.bytes_sent / 1e6
+
     def collect_metrics(self):
         # Get network usage metrics.
         network_usage = psutil.net_io_counters()
-        self._metrics["network_receive_megabytes"].append(network_usage.bytes_recv / 1e6)
-        self._metrics["network_transmit_megabytes"].append(network_usage.bytes_sent / 1e6)
+        # Usage metrics will be the diff between current and initial metrics.
+        self._metrics["network_receive_megabytes"] = (
+            network_usage.bytes_recv / 1e6 - self._initial_receive_megabytes
+        )
+        self._metrics["network_transmit_megabytes"] = (
+            network_usage.bytes_sent / 1e6 - self._initial_transmit_megabytes
+        )
+
+    def aggregate_metrics(self):
+        # Network metrics don't need to be averaged.
+        return dict(self._metrics)
```

Comment:
i now understand why we removed the `append` above :)

i find this pattern to be a bit confusing. in this case, `BaseMetricsMonitor.aggregate_metrics()` averages the metrics, but `NetworkMonitor.aggregate_metrics()` returns the raw metrics without any transformation. someone using these classes might be surprised by the difference in behavior.

maybe we should follow the a "composition over inheritance" approach. instead of having `BaseMetricsMonitor.aggregate_metrics()` provide an averaging operation, it could be designed in a more generic manner, such as simply returning metrics without any operation (just like in the `NetworkMonitor`). then, any specific metric monitor class (derived class) can override this method and implement its own aggregation operation as needed. 

WDYT @chenmoneygithub ?